src/transformState.js

```diff
@@ -4,8 +4,29 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action['type']) {
+      case 'addProperties':
+        Object.assign(state, action['extraData']);
```

Comment:
[CODE KNOWLEDGE]: Remember, if property key is a variable - use brackets object[key]. If it's called key - use dot access object.key.